### PR TITLE
Archiving: expose status flags from API

### PR DESCRIPTION
**Pull request by luontola**
#806

### CODE REVIEW

File: test/clj/rems/test/api/licenses.clj [hunk 1]
```diff
@@ -26,7 +26,7 @@
                      assert-response-is-ok
                      read-body)]
         (is (coll-is-not-empty? data))
-        (is (= #{:id :start :end :licensetype :title :textcontent :localizations :attachment-id} (set (keys (first data)))))))
+        (is (:id (first data)))))
```
Comment (Macroz) [inline test/clj/rems/test/api/licenses.clj]: Why not actually test here what the API returns?

File: test/clj/rems/test/api/resources.clj [hunk 2]
```diff
@@ -12,14 +12,13 @@
...
     (testing "get"
-      ;; just a basic smoke test for now
       (let [data (-> (request :get "/api/resources")
                      (authenticate api-key user-id)
                      app
                      assert-response-is-ok
                      read-body)]
         (is (coll-is-not-empty? data))
-        (is (= #{:id :owneruserid :modifieruserid :organization :resid :start :end :active :licenses} (set (keys (first data)))))))
+        (is (:id (first data)))))
```
Comment (Macroz) [inline test/clj/rems/test/api/resources.clj]: See above about testing the API.

File: src/clj/rems/db/catalogue.clj [hunk 3]
```diff
@@ -22,7 +22,10 @@
   "Associates localisations into a catalogue item from
   the preloaded state."
   [item]
-  (assoc item :localizations ((cached :localizations) (:id item))))
+  (-> item
+      (assoc :localizations ((cached :localizations) (:id item))
+             :end (:endt item))
```
Comment (Macroz) [inline src/clj/rems/db/catalogue.clj]: We could rename the SQL column to `end` so we don't need this?

File: src/clj/rems/db/resource.clj [hunk 4]
```diff
@@ -1,9 +1,7 @@
 (ns rems.db.resource
   (:require [rems.db.core :as db]
-            [rems.InvalidRequestException]
             [rems.util :refer [getx-user-id]])
-  (:import (org.postgresql.util PSQLException)
-           (rems InvalidRequestException)))
+  (:import (org.postgresql.util PSQLException)))
```
Comment (Macroz) [inline src/clj/rems/db/resource.clj]: I prefer using `[]` inside the import like in requires.

File: test/clj/rems/test/api/licenses.clj (hunk 1 above)
Comment (luontola) [reply test/clj/rems/test/api/licenses.clj]: It is already tested by the schema validation. This test only duplicates that (and does it less thoroughly). It’s enough to check the precense of one field to know that the API is not returning a schema validation error or other junk.

File: src/clj/rems/db/catalogue.clj (hunk 3 above)
Comment (luontola) [reply src/clj/rems/db/catalogue.clj]: Should endt be renamed to end in all tables? It’s in quite many places.

Comment (luontola) [reply src/clj/rems/db/catalogue.clj]: End is a reserved word 
https://www.postgresql.org/docs/8.1/sql-keywords-appendix.html

File: src/clj/rems/db/resource.clj (hunk 4 above)
Comment (luontola) [reply src/clj/rems/db/resource.clj]: It's not worth the effort to change. The IDE adds imports automatically and the established convention in this project is to use `()` - there are 23 instances of `(:import (` and only 3 instances of `(:import [`.

File: src/clj/rems/db/catalogue.clj (hunk 3 above)
Comment (Macroz) [reply src/clj/rems/db/catalogue.clj]: Yes it is but there is no need to avoid it . You can write `"end"`

Comment (Macroz) [reply src/clj/rems/db/catalogue.clj]: The rename of endt can be postponed to the cleaning phase next week if you add a task for it.

File: src/clj/rems/db/resource.clj (hunk 4 above)
Comment (Macroz) [reply src/clj/rems/db/resource.clj]: Then I will do the change to all of them myself. They are obviously someone else's results :)

File: test/clj/rems/test/api/licenses.clj (hunk 1 above)
Comment (Macroz) [reply test/clj/rems/test/api/licenses.clj]: In general there is value to overlapping test coverage as long as it's not too tedious to maintain. Especially having some through the API stack (most valuable happy paths) testing which is what the API tests are mostly about.

Also why not remove all the coll-`is-not-empty?` then?

Comment (luontola) [reply test/clj/rems/test/api/licenses.clj]: Each test has a cost - the time it takes to run it, the time it takes to understand why the test failed, and the time it takes to fix the test when it fails for reasons unrelated to what is the focus of the test. That's why tests should be written with as much care as the production code.

Removing `coll-is-not-empty?`

File: src/clj/rems/api/forms.clj [hunk 5]
```diff
@@ -35,13 +37,15 @@
   [Form])
 
 (defn- format-form
-  [{:keys [id organization title start endt active?]}]
+  [{:keys [id organization title start endt active? enabled archived]}]
```
[the remark this answers is not in the document]
Comment (luontola) [reply src/clj/rems/api/forms.clj]: Adding the question marks.

File: src/clj/rems/db/catalogue.clj (hunk 3 above)
Comment (luontola) [reply src/clj/rems/db/catalogue.clj]: I added a note to the ideas.

File: src/clj/rems/api/forms.clj (hunk 5 above)
[the remark this answers is not in the document]
Comment (luontola) [reply src/clj/rems/api/forms.clj]: On a second thought, removing the question mark from `active?` is much easier.

File: resources/sql/queries.sql [hunk 6]
```diff
@@ -3,7 +3,7 @@
 -- - Get catalogue items
 -- - :items vector of item ids
 -- - :resource resource id to fetch items for
-SELECT ci.id, ci.title, res.resid, ci.wfid, ci.formid, ci.start, ci.endt, ci.enabled, ci.archived
+SELECT ci.id, ci.title, res.resid, ci.wfid, ci.formid, ci.start, ci.endt as "end", ci.enabled, ci.archived
```
Comment (Macroz) [inline resources/sql/queries.sql]: This does not clean up the database however? Was it your intention?

File: src/clj/rems/api/forms.clj [hunk 7]
```diff
@@ -37,13 +37,13 @@
   [Form])
 
 (defn- format-form
-  [{:keys [id organization title start endt active? enabled archived]}]
+  [{:keys [id organization title start endt active enabled archived]}]
```
Comment (Macroz) [inline src/clj/rems/api/forms.clj]: There's still endt here?

File: resources/sql/queries.sql (hunk 6 above)
Comment (luontola) [reply resources/sql/queries.sql]: It's an old field. Better to rename it physically in a separate PR. It smells like an invasive operation.

File: src/clj/rems/api/forms.clj (hunk 7 above)
Comment (luontola) [reply src/clj/rems/api/forms.clj]: And in all other old code. It's a very invasive refactoring, so better do it in a separate PR. There's a note for it in the ideas backlog.

File: src/clj/rems/api/licenses.clj [hunk 8]
```diff
@@ -48,7 +48,7 @@
       :roles #{:owner}
       :query-params [{active :- (describe s/Bool "filter active or inactive licenses") nil}]
       :return Licenses
-      (ok (licenses/get-all-licenses (when-not (nil? active) {:active? active}))))
+      (ok (licenses/get-all-licenses (when-not (nil? active) {:active active}))))
```
Comment (Macroz) [inline src/clj/rems/api/licenses.clj]: You can simply write this?
```
(ok (licenses/get-all-licenses (when active {:active active}))))
```

File: resources/sql/queries.sql (hunk 6 above)
Comment (Macroz) [reply resources/sql/queries.sql]: Yes that's why I suggested a separate task for it.

File: src/clj/rems/api/licenses.clj (hunk 8 above)
Comment (luontola) [reply src/clj/rems/api/licenses.clj]: Done.

**Pull request merged**